src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala

```diff
@@ -53,12 +53,13 @@ final case class AuthorizationApiServiceImpl(
   ): Route = {
     val result: Future[(SessionToken, RateLimitStatus)] = for {
       (sessionClaims, roles, selfcareId) <- readJwt(identityToken).toFuture
-      tenantId                           <- getTenantIdOr(selfcareId)(upsertTenantBySelfcareId(selfcareId))
-      rateLimitStatus                    <- rateLimiter.rateLimiting(tenantId)
+      internalContexts                   <- generateInternalTokenContexts(interopTokenGenerator, sessionClaims)
+      tenantId <- getTenantIdOr(selfcareId)(upsertTenantBySelfcareId(selfcareId)(internalContexts))(internalContexts)
+      rateLimitStatus <- rateLimiter.rateLimiting(tenantId)
       customClaims: Map[String, String] = Map(
-        USER_ROLES            -> roles.toString(),
-        ORGANIZATION_ID_CLAIM -> tenantId.toString(),
-        SELFCARE_ID_CLAIM     -> selfcareId.toString()
+        USER_ROLES            -> roles,
+        ORGANIZATION_ID_CLAIM -> tenantId.toString,
+        SELFCARE_ID_CLAIM     -> selfcareId
```

Comment:
The `.toString()` is mandatory, the lack of it caused a bug some days ago

Reply:
it is mandatory only on UUID, Strings can be passed as they are